vendor/github.com/manicminer/hamilton/environments/apis.go

```diff
@@ -0,0 +1,97 @@
+package environments
+
+type ApiEndpoint string
+
+const (
+	AadGraphGlobalEndpoint  ApiEndpoint = "https://graph.windows.net"
+	AadGraphGermanyEndpoint ApiEndpoint = "https://graph.cloudapi.de"
+	AadGraphChinaEndpoint   ApiEndpoint = "https://graph.chinacloudapi.cn"
+	AadGraphUSGovEndpoint   ApiEndpoint = "https://graph.microsoftazure.us"
+	MsGraphGlobalEndpoint   ApiEndpoint = "https://graph.microsoft.com"
+	MsGraphGermanyEndpoint  ApiEndpoint = "https://graph.microsoft.de"
+	MsGraphChinaEndpoint    ApiEndpoint = "https://microsoftgraph.chinacloudapi.cn"
+	MsGraphUSGovL4Endpoint  ApiEndpoint = "https://graph.microsoft.us"
+	MsGraphUSGovL5Endpoint  ApiEndpoint = "https://dod-graph.microsoft.us"
+	MsGraphCanaryEndpoint   ApiEndpoint = "https://canary.graph.microsoft.com"
+)
+
+type ApiCliName string
+
+const (
+	AadGraphCliName ApiCliName = "aad-graph"
+	MsGraphCliName  ApiCliName = "ms-graph"
+)
+
+// API represent an API configuration for Microsoft Graph or Azure Active Directory Graph.
+type Api struct {
+	// The Application ID for the API.
+	AppId ApiAppId
+
+	// The Azure CLI codename for the API. Used with `az account get-access-token`.
+	CliName ApiCliName
```

Comment:
Ah thanks good to bear in mind. This needs further work anyway since the "codename" is defined in the `environments` package but isn't yet consumed by the `auth` package, which currently just uses an iota to toggle either API for CLI authentication.